### PR TITLE
Add estimated quote validation

### PR DESCRIPTION
**Pull request by pedromcunha**

### CODE REVIEW

File: packages/sdk/src/actions/call.ts [hunk 1]
```diff
@@ -55,6 +56,7 @@ function isSimulateContractRequest(tx: any): tx is SimulateContractRequest {
  * @param data.wallet Wallet object that adheres to the AdaptedWakket interface or a viem WalletClient
  * @param data.originChainId The chain to pay the solver on
  * @param data.precheck Set to true to skip executing steps and just to get the initial steps required
+ * @param data.estimatedQuote Set this property to check if
```
Comment (ted-palmer) [inline packages/sdk/src/actions/call.ts:59]: Need to finish param description

File: packages/sdk/src/utils/compareQuotes.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import type { Execute } from '../types/index.js'
+
+export const compareQuotes = (estimatedQuote: Execute, newQuote: Execute) => {
+  const estimatedIsSplit = (estimatedQuote.breakdown ?? []).length > 1
+  const newIsSplit = (newQuote.breakdown ?? []).length > 1
+
+  if (estimatedIsSplit != newIsSplit && newIsSplit) {
```
Comment (ted-palmer) [inline packages/sdk/src/utils/compareQuotes.ts:7]: What is the desired behavior if the expected quote goes from split route to single route? Do we still want to throw an error?

File: packages/sdk/src/utils/executeSteps.ts [hunk 3]
```diff
@@ -43,6 +43,7 @@ export async function executeSteps(
       gasLimit?: string
     }
   },
+  estimatedQuote?: Execute,
```
Comment (ted-palmer) [inline packages/sdk/src/utils/executeSteps.ts:46]: Feels like maybe this should be an action? Like `checkExpectedQuote`?

**Review by ted-palmer — CHANGES_REQUESTED** [review]

File: packages/sdk/src/actions/call.ts (hunk 1 above)
Comment (pedromcunha) [reply packages/sdk/src/actions/call.ts:59]: hah, thanks got distracted on that one.

**Pull request closed without merging**